### DISCUSSION TITLE
useFetch crashes nuxt 3 on page reload in build mode

### DISCUSSION THREAD

**Discussion by fabiomigliaccio** [nuxt/nuxt#17184] [Questions] [answered]
Hi,
I'm new to Nuxt, so maybe I'm doing something wrong; maybe it's a bug.
I'm using the 3.0.0-rc.6 version, everything works well in dev mode. 
However, launching the built app with `node .output/server/index.mjs ` results in a strange behavior: if I **reload any page that uses useFetch** the app crashes with this 500 error: 
```
[nuxt] [request error] nuxt instance unavailable
  at useNuxtApp (./.output/server/node_modules/nuxt/dist/app/nuxt.mjs:129:13)  
  at useAsyncData (./.output/server/node_modules/nuxt/dist/app/composables/asyncData.mjs:24:16)  
  at useFetch (./.output/server/node_modules/nuxt/dist/app/composables/fetch.mjs:46:21)  
  at ./.output/server/chunks/app/index.e09e69f2.mjs:31:89  <-- this is my page where useFetch is used
  ... other not important pages ...
```
To rule out that the problem could be the API itself it now returns a simple JSON object, but nothing changes:
```
export default eventHandler(event => {
    return [{"title":"xxx","slug":"yyy", ...}, { ... }]
});
```
This is how I use useFetch in the code:
```
<script setup>
  import {ref} from "vue";
  import {useFetch} from "nuxt/app";
  let categories = ref([])
  const { data: result }  = await useFetch('/api/json')
  categories.value = result.value
</script>
```
What's wrong?
Thanks in advance

**Comment by k0mar12** [comment, marked as answer]
Probably, you dont need to import `useFetch`. Remove import and it should works fine.

**Comment by jhisham** [comment]
In addition to the answer from @k0mar12, you don't have to explicitly import `ref` as well. This is because these modules are auto imported by Nuxt 3. Per the Nuxt 3 documentation:

> Nuxt auto-imports helper functions, composables and Vue APIs to use across your application without explicitly importing them. Based on the directory structure, every Nuxt application can also use auto-imports for its own components, composables and plugins. Components, composables or plugins can use these functions.

So any Vue APIs and files from your `components`, `composables`, and `plugins` also don't need to be explicitly imported.

If you would however like to explicitly import theses, you should use the `#imports` alias instead. Per the documents:

> Nuxt exposes every auto-import with the #imports alias that can be used to make the import explicit if needed:

```
<script setup>
  import { ref, computed } from '#imports'

  const count = ref(1)
  const double = computed(() => count.value * 2)
</script>
```
